src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp

```diff
@@ -3329,6 +3345,23 @@ CTranslatorRelcacheToDXL::IsIndexSupported(Relation index_rel)
 			GIN_AM_OID == index_rel->rd_rel->relam);
 }
 
+//---------------------------------------------------------------------------
+//	@function:
+//		CTranslatorRelcacheToDXL::IsIndexVisible
+//
+//	@doc:
+//		Check if index is visible in current transaction
+//
+//---------------------------------------------------------------------------
+BOOL
+CTranslatorRelcacheToDXL::IsIndexVisible(Relation index_rel)
+{
+	return !(index_rel->rd_index->indcheckxmin &&
+			 !TransactionIdPrecedes(
```

Comment:
I think we need to make gpdb:: wrappers over this functions, in order to unify the approach.
See the [IsIndexSupported](https://github.com/arenadata/gpdb/blob/d10d2bdea3f7a6df667d8fba60dd3d861f077883/src/backend/gpopt/translate/CTranslatorRelcacheToDXL.cpp#L3342) example